be/src/vec/core/block.h

```diff
@@ -264,9 +265,11 @@ class Block {
 
     void append_block_by_selector(MutableBlock* dst, const IColumn::Selector& selector) const;
 
+    // need exception safety
     static void filter_block_internal(Block* block, const std::vector<uint32_t>& columns_to_filter,
```

Comment:
it's better to modify this method to be exception safe, and return error status.
then we only catch exception inside this method. The callers only need to return_if_error